core/src/wallet/cosmos/explorers/StargateGaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -0,0 +1,1310 @@
+                cosmos::Validator val;
+                // Stargate explorer does not fill distInfo as it is too costly in HTTP calls
+                val.distInfo.selfBondRewards = "n/a";
+                val.distInfo.validatorCommission = "n/a";
+                parseValidatorSetEntry(validatorNode, val);
+                return Future<cosmos::Validator>::successful(val);
+            })
+        .template flatMap<cosmos::Validator>(
+            getContext(),
+            [this, parseJsonNumbersAsStrings](
+                const cosmos::Validator &inputVal) -> Future<cosmos::Validator> {
+                auto retval = cosmos::Validator(inputVal);
+                auto consAddr = CosmosLikeExtendedPublicKey::fromBech32(
+                                    _currency, retval.consensusPubkey, std::string(""))
+                                    ->derive("")
+                                    ->toBech32();
+                return _http
+                    ->GET(fmt::format(kGrpcSignInfoEndpoint, _ns, _version,
+                                      consAddr))
+                    .json(parseJsonNumbersAsStrings)
+                    .template flatMap<cosmos::Validator>(
+                        getContext(),
+                        [retval](
+                            const HttpRequest::JsonResult &response) mutable
+                        -> Future<cosmos::Validator> {
+                          const auto &document =
+                              std::get<1>(response)->GetObject();
+                          if (!document.HasMember(kValSignInfos)) {
+                            return Future<cosmos::Validator>::successful(
+                                retval);
+                          }
+                          parseSignInfo(document, retval.signInfo);
+                          return Future<cosmos::Validator>::successful(retval);
+                        });
+            });
+}
+
+FuturePtr<std::vector<cosmos::Delegation>> StargateGaiaCosmosLikeBlockchainExplorer::getDelegations(
+    const std::string &delegatorAddr) const
+{
+    const auto parseNumbersAsStrings = true;
+    const auto ignoreFailStatusCode = true;
+    return _http
+        ->GET(fmt::format(kGrpcDelegationsEndpoint, _ns, _version, delegatorAddr), ACCEPT_HEADER)
+        .json(parseNumbersAsStrings, ignoreFailStatusCode)
+        .mapPtr<std::vector<cosmos::Delegation>>(
+            getContext(), [this](const HttpRequest::JsonResult &response) {
+                // Catching http errors because Cosmos code returns 500 when there is no delegation.
+                if (std::get<0>(response)->getStatusCode() >= 300) {
+                     return std::make_shared<std::vector<cosmos::Delegation>>();
+                }
+                const auto &document = std::get<1>(response)->GetObject();
+                auto delegations = std::make_shared<std::vector<cosmos::Delegation>>();
+                // Handle null
+                if (document.HasMember("message")
```

Comment:
Is it intended to check for "message" here? If so then the constant should be used.